src/Components/Components/src/BindConverter.cs

```diff
@@ -1377,6 +1394,36 @@ private static class FormatterDelegateCache
                 return (BindFormatter<T>)formatter;
             }
 
+            private static BindFormatter<T[]> MakeArrayFormatter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>()
+            {
+                var elementFormatter = Get<T>();
+
+                return FormatArrayValue;
+
+                string? FormatArrayValue(T[] value, CultureInfo? culture)
+                {
+                    if (value.Length == 0)
+                    {
+                        return "[]";
+                    }
+
+                    var builder = new StringBuilder("[\"");
+                    builder.Append(elementFormatter(value[0], culture));
+                    builder.Append('\"');
+
+                    for (var i = 1; i < value.Length; i++)
+                    {
+                        builder.Append(", \"");
+                        builder.Append(elementFormatter(value[i], culture));
```

Comment:
If `T` is `string`, do we need to JSON-encode the value? Could it contains a `"` character? Alternatively we could use a custom encoding if we wanted.

Sorry I didn't spot this before!